### PR TITLE
Allow RequestParam and FileParam to target class

### PR DESCRIPTION
**Pull request by flohw**
Hi,

To be consistent across the different available params and be able to place them on class, it requires to add `\Attribute::TARGET_CLASS` to the `RequestParam` and `FileParam`.

I added the annotation for compatibility too.

Hopefully this can be merged and release before #2400 as they may be incompatible.

Thanks

[edit]: Actions seems to be broken due to composer.json or lock file. Not sure how to resolve. Let me know if I can help on this too.

### CODE REVIEW

File: Controller/Annotations/RequestParam.php
Comment (mbabker) [inline Controller/Annotations/RequestParam.php]: You need to add this to the existing `@Target` annotation to fix the test errors (same for the `FileParam` class).  Double-check this against the `QueryParam` class' annotations.

Comment (flohw) [reply Controller/Annotations/RequestParam.php]: Some habit are easier to forget than others... :grin: 

Thank you.

**Review by deluxetom — APPROVED** [review]

**Review by mbabker — APPROVED** [review]

**Comment by flohw** [comment]
Hi @goetas 

Just saw you on the pr 2401 and was seeking for the maintainer. Hopefully I reached the right person to know how we could manage this PR. :slightly_smiling_face: 

I thought it would be nice to be merged before and maybe make a release (hopefully not too late because of the last merge :worried: ) to allow a more flexible usage of the current attributes.

Please let me know if I need to target another branch or if the improvement will be merged once SF7 compatibility is achieved.

Thank you.

**Comment by goetas** [comment]
Ooo, I saw this too late... Sorry 😔

**Comment by mbabker** [comment]
> Ooo, I saw this too late... Sorry 😔

I did copy it into #2400 so it's there.

**Comment by flohw** [comment]
No problem. Thank you.

I keep an eye on the bundle as a lot of people.
Let us know if we can help on some issue or pr. I haven't seen a "good first issue" or something.

I will close this pr once the other is merged or close it now if you think it's more appropriate.

**Comment by flohw** [comment]
I looked at your comment and pr on the phone and didn't see that the other pr was merged and a release was made.

Thank you for allowing the update! :slightly_smiling_face:

**Pull request closed without merging**
